src/models/st7789/variants.rs

```diff
@@ -48,6 +48,28 @@ where
             ST7789::new(ModelOptions::with_display_size(options, 240, 240)),
         )
     }
+    ///
+    /// Creates a new [Display] instance with [ST7789] as the [Model] with
+    /// general variant using display size of 240x240 but a frame buffer of 240x320 and adjusting the offset
+    ///
+    /// # Arguments
+    ///
+    /// * `di` - a [DisplayInterface](WriteOnlyDataCommand) for talking with the display
+    /// * `rst` - display hard reset [OutputPin]
+    /// * `options` - the [DisplayOptions] for this display/model
+    ///
+    pub fn st7789_240x240_b240x320(di: DI, rst: Option<RST>, options: DisplayOptions) -> Self {
```

Comment:
I'm wondering if the "general" https://github.com/almindor/mipidsi/blob/master/src/models/st7789/variants.rs#L26-L32 ST7789 constructor should be used here and just modified with the offsets.

Reply:
The display size woudn't be right if we used the general constructor, though the `st7789_240x240` constructor could be used, but then the frame buffer also wouldn't be right (though maybe it's fine since they shouldn't be used). 

Now as you said maybe it would be possible to use a separate form of offset but I am not familiar with the spec I would need to look it up but it seems the spec is for members only. If there is a cleaner way to do this I am definitely open to it.

Reply:
I thought about this a bit more, the issue with another offset setup is that it'd conflict with the clipping offsets.

E.g. I could use the HW scroll to offset the framebuffer "active area" in inverted orientations, and it would work here, but it would break with pico1 or others that have limited display sizes in combination with offset requirements under default conditions.

So I think I will accept this, but I want to test some more first.